api/web/base.py

```diff
@@ -310,6 +341,39 @@ def handle_exception(self, exception, debug):
 
         util.send_json_http_exception(self.response, str(exception), code, custom=custom_errors)
 
+    def log_user_access(self, access_type, cont_name=None, cont_id=None):
```

Comment:
I'm worried about the logging parameter taking optional arguments. For the `access_type` that require it, should this function throw if these opt args are missing?